src/addons/messagelog/src/main/java/ee/ria/xroad/proxy/messagelog/TaskQueue.java

```diff
@@ -86,6 +86,13 @@ protected void handleTimestampSucceeded(TimestampSucceeded message) {
         } finally {
             if (succeeded) {
                 indicateSuccess();
+                // If time-stamped records count equals to time-stamp records limit, there are probably
+                // still records to be time-stamped. Init another another time-stamping round to prevent
+                // messagelog records to begin to bloat.
+                if (message.getMessageRecords().length == MessageLogProperties.getTimestampRecordsLimit()) {
```

Comment:
```suggestion
                if (message.getMessageRecords().length >= MessageLogProperties.getTimestampRecordsLimit()) {
```

Reply:
The maximum number of records to be processed in one batch is set by `timestamp-records-limit` property. The property is used for setting max results count when the database query that's fetching the records to be time-stamped is created:

https://github.com/nordic-institute/X-Road/blob/develop/src/addons/messagelog/src/main/java/ee/ria/xroad/proxy/messagelog/TaskQueue.java#L189

```java
    private List<Task> getTimestampTasks(Session session) {
        return session.createQuery(getTaskQueueQuery()).setMaxResults(
                MessageLogProperties.getTimestampRecordsLimit()).list();
    }
```

Therefore, the number of records that are time-stamped in one batch cannot be greater than `timestamp-records-limit`.

Reply:
Yes, that's correct, ">=" would just be defensive programming.